store/tikv/lock_resolver.go

```diff
@@ -146,6 +149,79 @@ func (lr *LockResolver) getResolved(txnID uint64) (TxnStatus, bool) {
 	return s, ok
 }
 
+// BatchResolveLocks resolve locks in a batch
+func (lr *LockResolver) BatchResolveLocks(bo *Backoffer, locks []*Lock, loc RegionVerID) (bool, error) {
+	if len(locks) == 0 {
+		return true, nil
+	}
+
+	lockResolverCounter.WithLabelValues("resolve").Inc()
+
+	var expiredLocks []*Lock
+	for _, l := range locks {
+		if lr.store.GetOracle().IsExpired(l.TxnID, l.TTL) {
+			lockResolverCounter.WithLabelValues("expired").Inc()
+			expiredLocks = append(expiredLocks, l)
+		} else {
+			lockResolverCounter.WithLabelValues("not_expired").Inc()
+		}
+	}
+	if len(expiredLocks) == 0 {
+		log.Error("This should not occur!")
+		return true, nil
+	}
+
+	lr.cntTxnStatus = 0
+	lr.costTxnStatus = 0
```

Comment:
Not suitable to use a struct field here, because the `LockResolver` could be accessed by multiple goroutines.

Reply:
I think there is no need to do it here, I will move then out of the function.